mage_integrations/mage_integrations/destinations/kafka/__init__.py

```diff
@@ -0,0 +1,66 @@
+import argparse
+import json
+import sys
+from typing import Dict, List
+
+import pandas as pd
+from kafka import KafkaProducer
+
+from mage_integrations.destinations.base import Destination
+from mage_integrations.destinations.constants import KEY_RECORD
+from mage_integrations.destinations.utils import update_record_with_internal_columns
+from mage_integrations.utils.dictionary import merge_dict
+
+MAXIMUM_BATCH_SIZE_MB = 1
+
+
+class Kafka(Destination):
+
+    def test_connection(self) -> None:
+        raise Exception('Not Implemented')
+
+    def build_client(self):
+        kwargs = dict(
+            bootstrap_servers=self.config['bootstrap_server'],
+            api_version=self.config.get('api_version', None),
+            value_serializer=lambda x: json.dumps(x).encode('utf-8'),
+            key_serializer=lambda x: x.encode('utf-8') if x else None,
+        )
+        producer = KafkaProducer(**kwargs)
+        return producer
+
+    def export_batch_data(self, record_data: List[Dict], stream: str, tags: Dict = None) -> None:
+
+        self.logger.info('Export data started.')
+
+        for r in record_data:
+            r['record'] = update_record_with_internal_columns(r['record'])
+
+        df = pd.DataFrame([d[KEY_RECORD] for d in record_data])
+        df_count = len(df.index)
+
+        idx = 0
+        total_byte_size = int(df.memory_usage(deep=True).sum())
+        tags2 = merge_dict(tags, dict(
+            total_byte_size=total_byte_size,
+        ))
+
+        self.logger.info(f'Inserting records for batch {idx} started.', tags=tags2)
+
+        value = df.to_json()
+
+        producer = self.build_client()
+        producer.send(self.config['topic'], value, key=self.config.get('key', None))
```

Comment:
You should send the records as different messages instead of the big json.

Reply:
You don't need to convert the records into dataframe. Just loop through the records and write them to kafka.

Reply:
Also, you can get the key column from the catalog.

Reply:
Removed all DataFrame creation and set each record formatted as a JSON by default
Example:
![Screenshot from 2023-11-21 02-16-56](https://github.com/mage-ai/mage-ai/assets/14100959/100ddcd5-2b9d-495c-a93e-432fa88897ea)

Also updated the docs to explain in simple terms how the key column and message handling now works